Wdb 62/configure password encryption

kibartas opened the pull request:
On to implementing the Login endpoint

Commit:
[WDB-62]
add  row to User entity

Commit:
[WDB-62]
create password hash function

Commit:
[WDB-62]
create test endpoint for the password hash function

AArlauskas commented on backend/backend/Models/User.cs:
```diff
@@ -23,6 +23,9 @@ public User()
         [Required]
         public string Password { get; set; }
+        
+        [Required]
```
Please, kind sir, could you add salting to the constructor?

kibartas replied:
I'd be glad to

Commit:
[WDB-62]
change test controller method

Commit:
[WDB-62]
remove set from Salt and Password properties and change constructor

Commit:
[WDB-62]
add a private constructor to User; set some properties 'set' to 'private set' so they couldn't be reached on accident; add new method to the User entity setNewPassword

Commit:
[WDB-62]
augment hashing test endpoint

Commit:
[WDB-62]
add 'CheckPlaintextAgainstHash' helper method to Hasher class

AArlauskas reviewed (approved):
Nice job Salt bae

salmonela7 reviewed (approved):
Looks good. 👍

Commit:
[WDB-62]
merge master and remove test endpoint for hash function
